### DISCUSSION TITLE
GroupDataLoader many to many functionality (bug?)

### DISCUSSION THREAD

**Discussion by Azagatoth** [ChilliCream/hotchocolate#4828] [Q&A]
I have three tables in my database that are connected as a many to many relation.
They are called Asset, MeterPoint and AssetMeterPointMember.

When performing this query I get an unexpected result:

```
query
{
	meterPoints
	{
		id,
		eic,
		assetMeterPointMembers (79 keys)
		{
			asset (80 keys)
			{
				id,
				assetMeterPointMembers (13 keys)
				{
					meterPoint (46 keys)
					{
						id,
						eic
					}
				}
			}
		}
	}
}
```

**Database**

There are 79 MeterPoints.

53 of those have Assets connected through AssetMeterPointMember.

There are 88 Assets and 80 of those are connected to a MeterPoint.

**GraphQl**

Keys = primary keys for the database.

-> = Are sent to GroupDataLoader.

79 keys -> AssetMeterPointMember GroupDataLoader in MeterPoint (correct)

80 keys -> Asset GroupDataLoader in AssetMeterPointMember (correct)

13 keys -> AssetMeterPointMember GroupDataLoader in Asset (incorrect)

The 13 keys are always fetched from the newest data, but there are some keys left out, so no logic there.

46 keys -> MeterPoint GroupDataLoader from AssetMeterPointMember (incorrect)

**Expected**

I would expect to have at least the same amount of keys going up again in the hierarchy.

Can someone please explain what might cause this.

**Comment by PascalSenn** [comment]
@Azagatoth  DataLoader only fetch the data once based on the ID. 
Could it be that the "missing" keys are already fetched in the first dataloader call?

**Reply by Azagatoth** [reply]
That is a good point, I will investigate this tomorrow and mark your reply if this is the case

**Reply by Azagatoth** [reply]
Sorry, but this is not the case... There is an overlap of all 13 keys.
Which means the 80 keys from the first Asset GroupDataLoader also contain the 13 keys from the AssetMeterPointMember when traversing up again.
